application/src/pages/Registration/Registration.tsx

```diff
@@ -0,0 +1,101 @@
+import { Formik, FormikErrors } from 'formik';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { redirect } from '../../bll/redirect';
+import { setEmailPassword } from '../../bll/reducers/registrationReducer';
+import { select } from '../../bll/select';
+import { PATH } from '../PageRoutes/PageRoutes';
+import s from './Registration.module.css';
+
+type FormValues = {
+  email: string;
+  password: string;
+};
+export const Registration: React.FC = () => {
+  redirect();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { password, email } = useSelector(select);
+
+  useEffect(() => {
+    if (password && email) {
+      navigate(PATH.PROFILE);
+    }
+  }, [password, email]);
+
+  return (
+    <div className={s.main}>
+      <div className={s.table}>
+        <h1>Registration page</h1>
+        <Formik
+          initialValues={{
+            email: '',
+            password: ''
+          }}
+          validate={(values) => {
+            const errors: FormikErrors<FormValues> = {};
+            if (!values.email) {
+              errors.email = 'Required';
+            } else if (!values.password) {
+              errors.password = 'Required';
+            } else if (
+              !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
```

Comment:
It is better to write this regex to a variable.
const VALIDATE_EMAIL = new RegExp('^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$', 'i');